components/camera/client.go

```diff
@@ -316,6 +316,11 @@ func (c *client) Properties(ctx context.Context) (Properties, error) {
 	}
 	result.MimeTypes = resp.MimeTypes
 	result.SupportsPCD = resp.SupportsPcd
+
+	// Check if the optional frame_rate is present and set it if it exists
+	if resp.FrameRate != nil {
```

Comment:
We need this check because if we attempt dereference the pointer when nil we will get a panic?

Reply:
Yes - we will get undefined behavior if we dereference a null pointer so that is why i put that check in place.